CHANGELOG.md

```diff
@@ -19,6 +19,9 @@ This component adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.h
 
 ### Fixed
 
+- Deadlock: AssemblyResolve_ManagedProfilerDependencies should be moved out
```

Comment:
- Fixed deadlock occurring in Assembly Resolver.
  See [#4269](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/4269)
  for details.